### PR TITLE
Migrating to Github Actions - Tests

### PR DESCRIPTION
**Pull request by sadrasabouri**
#### Reference Issues/PRs
#361 

+ `pythonpublish.yml` renamed to `publish.yml`
+ `test.yml` file added
+ Travis test files removed
+ Appveyor test files removed
+ `requirement_splitter.py` added

### CODE REVIEW

**Review by sepandhaghighi — APPROVED** [review]
LGTM 💯

**Comment by sepandhaghighi** [comment]
@alirezazolanvari Please review this PR.
If everything is okay, please merge this PR after approval.

**Review by alirezazolanvari — APPROVED** [review]

**Pull request merged**
